### PR TITLE
Merge convenience layer into protocol layer

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
I merged the code with as few changes as possible, however there is some opportunity for further refinement (e.g. no longer needing to pass global client params).  Will address in a separate PR.

### CODE REVIEW

File: src/generator/generator.ts [hunk 1]
```diff
@@ -7,40 +7,45 @@ import { serialize } from '@azure-tools/codegen';
...
-
-    const operations = await generateOperations(session);
     for (const op of values(operations)) {
       host.WriteFile(`${op.name.toLowerCase()}.go`, op.content, undefined, 'source-file-go');
     }
 
     const enums = await generateEnums(session);
     if (enums.length > 0) {
-      host.WriteFile('enums.go', enums, undefined, 'source-file-go');
+      host.WriteFile(`enums.go`, enums, undefined, 'source-file-go');
```
Comment (catalinaperalta) [inline src/generator/generator.ts]: Why did you change to these types of quotes? According to me the correct thing is to use ' ' if it's a simple string as above.

File: src/generator/generator.ts [hunk 2]
```diff
@@ -7,40 +7,45 @@ import { serialize } from '@azure-tools/codegen';
...
     }
 
     const models = await generateModels(session);
-    host.WriteFile('models.go', models, undefined, 'source-file-go');
+    host.WriteFile(`models.go`, models, undefined, 'source-file-go');
```
Comment (catalinaperalta) [inline src/generator/generator.ts]: Same as my previous comment

File: src/generator/generator.ts (hunk 1 above)
Comment (jhendrixMSFT) [reply src/generator/generator.ts]: It was from copy/paste from the other layer, will fix.
